src/util/mpir_pmi.c

```diff
@@ -359,6 +359,14 @@ int MPIR_pmi_barrier(void)
     return mpi_errno;
 }
 
+int MPIR_pmi_barrier_only(void)
```

Comment:
This is fine for now. But later we probably want to rename this back the `MPIR_pmi_barrier` and rename the previous `MPIR_pmi_barrier` into `MPIR_pmi_fence`.